src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/controller/ServiceClientHelper.java

```diff
@@ -104,4 +105,27 @@ public BadRequestException wrapInBadRequestException(
         return new BadRequestException(INVALID_SERVICE_CLIENT_ID + e.getServiceClientIdentifier(),
                 new ErrorDeviation(ERROR_INVALID_SERVICE_CLIENT_ID));
     }
+
+    /**
+     * Sort a list of ServiceClient objects using member name / group description as the primary sort key, and client id
+     * as the secondary sort key.
+     * @param clients
+     */
+    public void sortServiceClientsList(List<ServiceClient> clients) {
```

Comment:
Instead of this concrete list-sorting method (and `ClientUtils.sortClientsList`) a `Comparator` implementing the comparison logic would be more reusable (can sort arrays, sorted sets etc), and maybe even a bit cleaner design. Both Comparators could then be in same package (probably org.niis.xroad.restapi.controller or org.niis.xroad.restapi.openapi - or maybe even a new org.niis.xroad.restapi.converter.comparator to signal this is part of the "converting" domain...) and so be more consistent with each other and "discoverable".

For the actual implementation, there's also alternatives https://stackoverflow.com/a/20093642/1469083 such as 
- using Comparator.comparing+thenComparing+nullsFirst builders (end result may be cleaner or messier)
- using Guava or Apache commons helpers

Although the actual implementation is pretty simple so I don't know if alternative ways are worth it.

Reply:
Created a new package `org.niis.xroad.restapi.converter.comparator` and added both comparators implementing the `Comparator` interface under it. In this way they are more discoverable. However, the implementation was left as it was before.